src/tests/ftest/util/dmg_utils.py

```diff
@@ -427,125 +427,61 @@ def pool_create(self, scm_size, uid=None, gid=None, nvme_size=None,
-                "disabled": 2,
-                "leader": 3,
-                "version": 4
-            }
-            # Mapping of the pool space entries to the match[2|3] indices
-            space_map = {
-                "total": 6,
-                "free": 7,
-                "free_min": 8,
-                "free_max": 9,
-                "free_mean": 10
-            }
-            # Mapping of the second indices mappings to the first match indices
-            map_values = {
-                0: pool_map,
-                1: {"target_count": 5},
-                2: space_map,
-                3: space_map,
-                4: {"status": 11, "status2": 12, "status3": 13}
-            }
-            for index_1, match_list in enumerate(match):
-                if index_1 not in map_values:
-                    continue
-                for key, index_2 in map_values[index_1].items():
-                    if index_1 == 2:
-                        if "scm" not in data:
-                            data["scm"] = {}
-                        data["scm"][key] = match_list[index_2]
-                    elif index_1 == 3:
-                        if "nvme" not in data:
-                            data["nvme"] = {}
-                        data["nvme"][key] = match_list[index_2]
-                    elif index_1 == 4:
-                        if "rebuild" not in data:
-                            data["rebuild"] = {}
-                        data["rebuild"][key] = match_list[index_2]
-                    else:
-                        data[key] = match_list[index_2]
-        return data
+        # Sample JSON output
+        # {
+        #     "response": {
+        #         "status": 0,
+        #         "uuid": "EDAE0965-7A6E-48BD-A71C-A29F199C679F",
+        #         "total_targets": 8,
+        #         "active_targets": 8,
+        #         "total_nodes": 1,
+        #         "disabled_targets": 0,
+        #         "version": 1,
+        #         "leader": 0,
+        #         "rebuild": {
+        #             "status": 0,
+        #             "state": "idle",
+        #             "objects": 0,
+        #             "records": 0
+        #         },
+        #         "scm": {
+        #             "total": 16000000000,
+        #             "free": 15999992320,
+        #             "min": 1999999040,
+        #             "max": 1999999040,
+        #             "mean": 1999999040
+        #         },
+        #         "nvme": {
+        #             "total": 32000000000,
+        #             "free": 31999950848,
+        #             "min": 3999993856,
+        #             "max": 3999993856,
+        #             "mean": 3999993856
+        #         }
+        #     },
+        #     "error": null,
+        #     "status": 0
+        # }
+        if use_json:
+            return self._get_json_result(("pool", "query"), pool=pool)
+
+        return self._get_result(("pool", "query"), pool=pool)
```

Comment:
This really should be just:
```suggestion
        return self._get_json_result(("pool", "query"), pool=pool)
```

Any callers of this method will just have need to implement this conversion:
```
data["uuid"]               -> result["response"]["uuid"]
data["ntarget"]            -> result["response"]["total_targets"]
data["disabled"]           -> result["response"]["disabled_targets"]
data["leader"]             -> result["response"]["leader"]
data["version"]            -> result["response"]["version"]
data["target_count"]       -> result["response"]["uuid"]
data["scm"]["total"]       -> result["response"]["scm"]["total"]
data["scm"]["free"]        -> result["response"]["scm"]["free"]
data["scm"]["free_min"]    -> result["response"]["scm"]["min"]
data["scm"]["free_max"]    -> result["response"]["scm"]["max"]
data["scm"]["free_mean"]   -> result["response"]["scm"]["mean"]
data["nvme"]["total"]      -> result["response"]["nvme"]["total"]
data["nvme"]["free"]       -> result["response"]["nvme"]["free"]
data["nvme"]["free_min"]   -> result["response"]["nvme"]["min"]
data["nvme"]["free_max"]   -> result["response"]["nvme"]["max"]
data["nvme"]["free_mean"]  -> result["response"]["nvme"]["mean"]
data["rebuild"]["status"]  -> result["response"]["rebuild"]["state"]
data["rebuild"]["status2"] -> result["response"]["rebuild"]["objects"]
data["rebuild"]["status3"] -> result["response"]["rebuild"]["records"]
```